### ISSUE TITLE
Crash when default start rule has parameters

### ISSUE THREAD

**Issue by pdubroy** [cdglabs/ohm#77]
```
> ohm.grammar('G { Start<x> = x }').match('x')
TypeError: Cannot read property 'getExprType' of undefined
    at Apply.pexprs.Apply._calculateExprType (/Users/dubroy/dev/cdg/ohm/src/pexprs-getExprType.js:112:28)
    at Apply.pexprs.PExpr.getExprType (/Users/dubroy/dev/cdg/ohm/src/pexprs-getExprType.js:33:19)
    at Apply.PExpr.newInputStreamFor (/Users/dubroy/dev/cdg/ohm/src/pexprs.js:39:23)
    at Object.Grammar._match (/Users/dubroy/dev/cdg/ohm/src/Grammar.js:102:38)
    at Object.Grammar.match (/Users/dubroy/dev/cdg/ohm/src/Grammar.js:93:22)
    at repl:1:35
    at REPLServer.defaultEval (repl.js:252:27)
    at bound (domain.js:287:14)
    at REPLServer.runBound [as eval] (domain.js:300:12)
    at REPLServer.<anonymous> (repl.js:417:12)
```

**Comment by pdubroy** [comment]
Not sure how we should handle this. I wonder if we should require that the default start rule for a grammar have no parameters.

**Comment by alexwarth** [comment]
> I wonder if we should require that the default start rule for a grammar have no parameters.

I like this idea. It wouldn't break anything that not currently broken anyway... Then the grammar above wouldn't have a default rule, which suggests that we should throw a "this grammar has no default start rule" error if you call `match` on it w/o specifying a start rule.

An alternative would be to introduce syntax to specify the default arguments to a parameterized rule -- I'm not sure how I feel about that yet. What's your take on it?

**Comment by pdubroy** [comment]
For now, I think it makes sense to throw an error if the default start rule requires parameters. If we end up implementing default arguments, then we should allow a parameterized rule to be the start rule as long as it has defaults for all the parameters.

**Comment by mroeder** [comment]
Will partially be solved - with an error message - by #76.

**Comment by mroeder** [comment]
After fixing #76, the error message you will get now is `Error: Wrong number of parameters for rule Start (expected 1, got 0)`.
You can run the match by explicitly calling `match('x', 'Start<"x">')`.
Does that fix and close this issue (for now)?
